src/transaction/TransactionController.test.ts

```diff
@@ -606,28 +606,23 @@ describe('TransactionController', () => {
   });
 
   it('should cancel a transaction', async () => {
-    await new Promise(async (resolve) => {
-      const controller = new TransactionController({ provider: PROVIDER });
-      controller.context = {
-        NetworkController: MOCK_NETWORK,
-      } as any;
-      controller.onComposed();
-      const from = '0xc38bf1ad06ef69f0c04e29dbeb4152b4175f0a8d';
-      const { result } = await controller.addTransaction({
-        from,
-        to: from,
-      });
-      controller.cancelTransaction('foo');
-      controller.hub.once(`${controller.state.transactions[0].id}:finished`, () => {
-        expect(controller.state.transactions[0].transaction.from).toBe(from);
-        expect(controller.state.transactions[0].status).toBe(TransactionStatus.rejected);
-      });
-      controller.cancelTransaction(controller.state.transactions[0].id);
-      result.catch((error) => {
-        expect(error.message).toContain('User rejected the transaction');
-        resolve('');
-      });
+    const controller = new TransactionController({ provider: PROVIDER });
+    controller.context = {
+      NetworkController: MOCK_NETWORK,
+    } as any;
+    controller.onComposed();
+    const from = '0xc38bf1ad06ef69f0c04e29dbeb4152b4175f0a8d';
+    const { result } = await controller.addTransaction({
+      from,
+      to: from,
+    });
+    controller.cancelTransaction('foo');
+    controller.hub.once(`${controller.state.transactions[0].id}:finished`, () => {
+      expect(controller.state.transactions[0].transaction.from).toBe(from);
```

Comment:
Nit: This was a pre-existing problem, but.... there is no guarantee of these lines running during the test :thinking: This is problematic.

We should wrap this in a Promise so that we can ensure it resolves before the test finishes.

Reply:
It appears it doesn't? :O 41eb582

Reply:
This is another "unintuitive order-of-execution" issue. They're not supposed to resolve until after everything else in the test